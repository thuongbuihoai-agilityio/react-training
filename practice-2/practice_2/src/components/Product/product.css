.product {
  position: relative;
}

.product__view {
  position: absolute;
  text-align: center;
  bottom: 0;
  margin: 0;
  width: 250px;
  padding: 10px 0;
  color: transparent;
  color: #333;
  background-color: #ffd89e;
  display: none;
}

.product__image {
  width: 250px;
  height: 300px;
}

.product__image:hover {
  opacity: 0.5;
}

.product__image:hover+.product__view {
  display: block;
}