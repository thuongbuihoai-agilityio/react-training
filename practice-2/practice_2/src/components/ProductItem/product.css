.product {
  position: relative;
}

.product__view {
  position: absolute;
  text-align: center;
  bottom: 0;
  margin: 0;
  width: 350px;
  padding: 10px 0;
  color: transparent;
  color: #333;
  background-color: #ffd89e;
  display: none;
  z-index: 5;
}

.product__link {
  color: black;
  text-decoration: none;
}

.product__image {
  width: 350px;
  height: 400px;
}

.product__image:hover {
  opacity: 0.5;
}

.product__image:hover+.product__view {
  display: block;
}