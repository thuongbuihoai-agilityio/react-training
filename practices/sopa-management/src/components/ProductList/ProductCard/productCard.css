.product-card {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  width: var(--w-extra-large);
  justify-content: space-between;
}
