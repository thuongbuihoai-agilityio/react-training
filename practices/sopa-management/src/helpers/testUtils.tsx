import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import {
  QueryClient,
  QueryClientProvider
} from 'react-query';
import {
  BrowserRouter,
  Router
} from 'react-router-dom';

export const renderRouterTest = (component: JSX.Element) => {
  const history = createMemoryHistory();
  return render(
    <Router location={history.location} navigator={history}>
      {component}
    </Router>
  );
};

export const renderWithRouterAndQuery = (children: JSX.Element) => {
  const queryClient = new QueryClient();

  return render(
    <BrowserRouter>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </BrowserRouter>
  );
};
