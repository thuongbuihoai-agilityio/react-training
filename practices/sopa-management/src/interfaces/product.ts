export interface Product {
  id: string;
  name: string;
  price: number;
  color: string;
  image: {
    url: string;
    alt: string;
  };
  size: string;
  description?: string;
  quantity: number;
};
