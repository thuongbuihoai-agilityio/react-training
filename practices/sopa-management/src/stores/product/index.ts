import { create } from 'zustand';
import { Product } from '@interfaces/product';

interface ProductType {
  products: Product[];
  setProducts: (value: Product[]) => void;
}

export const useProductStore = create<ProductType>()((set) => ({
  products: [],
  setProducts: (data: Product[]) => set(() => ({ products: data }))
}));
