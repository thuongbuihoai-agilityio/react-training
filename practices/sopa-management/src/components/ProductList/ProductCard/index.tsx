import { memo } from 'react';

// Components
import Card from '@components/Card';

// Interfaces
import { Product } from '@interfaces/product';

// Styles
import './productCard.css';

interface ProductCardProps {
  key?: number;
  data: Product[];
}

const ProductCard = ({
  data,
}: ProductCardProps) => (
  <div className='product-card' data-testid='product-card'>
    {data?.map((product) => (
      <Card
        key={product.id}
        href={`products/${product.id}`}
        src={product.image.url}
        name={product.name}
        color={product.color}
        price={product.price}
        alt={product.image.alt}
      />
    ))}
  </div>
);

export default memo(ProductCard);
