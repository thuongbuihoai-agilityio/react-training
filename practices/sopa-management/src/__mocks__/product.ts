import { Product } from "../interfaces/product";

const MOCK_PRODUCTS: Product[] = [
  {
    name: 'Black shoes',
    price: 87,
    size: 'Small',
    description:
      'Our goal in designing the super-soft Atoms Crew Socks was to provide a seamless match with your Atoms shoes, enhancing your comfort for everyday wear - making them the perfect socks for your everyday shoes.',
    color: 'Black',
    image: {
      url: '/images/products/black-shoes.png',
      alt: 'This is black shoes'
    },
    id: '1',
    quantity: 2
  },
  {
    name: 'Blue shoes',
    price: 45,
    size: 'Medium',
    description:
      'Our goal in designing the super-soft Atoms Crew Socks was to provide a seamless match with your Atoms shoes, enhancing your comfort for everyday wear - making them the perfect socks for your everyday shoes.',
    color: 'Blue',
    image: {
      url: '/images/products/blue-shoes.png',
      alt: 'This is blue shoes'
    },
    id: '2',
    quantity: 1
  }
];

const MOCK_PRODUCT: Product = {
  name: 'Black shoes',
  price: 87,
  size: 'Large',
  description: 'Our goal in designing the super-soft Atoms Crew Socks was to provide a seamless match with your Atoms shoes, enhancing your comfort for everyday wear - making them the perfect socks for your everyday shoes.',
  color: 'Black',
  image: {
    url: '/images/products/black-shoes.png',
    alt: 'This is black shoes'
  },
  id: '1',
  quantity: 3
};

export {
  MOCK_PRODUCTS,
  MOCK_PRODUCT
};
