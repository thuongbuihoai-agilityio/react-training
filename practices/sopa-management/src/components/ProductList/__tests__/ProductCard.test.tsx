// Libs
import '@testing-library/jest-dom';

// Components
import ProductCard from '../ProductCard';

// Helpers
import { renderRouterTest } from '@helpers/testUtils';

// Mocks
import { MOCK_PRODUCTS } from '@mocks/product';

describe('ProductList component', () => {
  test('should render ProductList component', () => {
    const { getByTestId } = renderRouterTest(
      <ProductCard data={MOCK_PRODUCTS} />
    );
    expect(getByTestId('product-card')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = renderRouterTest(
      <ProductCard data={MOCK_PRODUCTS} />
    );
    expect(container).toMatchSnapshot();
  });
});
