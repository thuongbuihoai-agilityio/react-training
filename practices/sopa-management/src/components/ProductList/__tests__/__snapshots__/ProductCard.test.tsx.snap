// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList component matches snapshot 1`] = `
<div>
  <div
    class="product-card"
    data-testid="product-card"
  >
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="card-item"
      >
        <div
          class="card-image"
        >
          <a
            aria-label="link to home page"
            data-testid="image-link"
            href="/products/1"
          >
            <img
              alt="This is black shoes"
              class="card-image-item"
              height="130"
              sizes=""
              src="/images/products/black-shoes.png"
              width="322"
            />
          </a>
        </div>
        <div
          class="card-info"
        >
          <a
            aria-label="Black shoes"
            class="card-name"
            href="/products/1"
          >
            Black shoes
          </a>
          <div
            class="card-description"
          >
            <p
              class=" text-highlight-primary"
              data-testid="text"
            >
              Black
            </p>
            <p
              class=" price-primary"
              data-testid="price"
            >
              $
              87
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="card-item"
      >
        <div
          class="card-image"
        >
          <a
            aria-label="link to home page"
            data-testid="image-link"
            href="/products/2"
          >
            <img
              alt="This is blue shoes"
              class="card-image-item"
              height="130"
              sizes=""
              src="/images/products/blue-shoes.png"
              width="322"
            />
          </a>
        </div>
        <div
          class="card-info"
        >
          <a
            aria-label="Blue shoes"
            class="card-name"
            href="/products/2"
          >
            Blue shoes
          </a>
          <div
            class="card-description"
          >
            <p
              class=" text-highlight-primary"
              data-testid="text"
            >
              Blue
            </p>
            <p
              class=" price-primary"
              data-testid="price"
            >
              $
              45
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
